### ISSUE TITLE
Column labels are not showing on IBMi / DB2 / AS400 JDBC driver

### ISSUE THREAD

**Issue by xantari** [dbeaver/dbeaver#13335]
#### System information: 
- Operating system (distribution) and version: 
Windows 10 / 64bit
- DBeaver version: 
21.0.1
- Additional extensions: 
None

#### Connection specification:
- Database name and version: 
com.ibm.as400.access.AS400JDBCDriver, jt400-10.6.jar (10.6) - Latest available as of this bug report
- Driver name: 
JT400 / com.ibm.as400.access.AS400JDBCDriver
- Do you use tunnels or proxies (SSH, SOCKS, etc)?
No

#### Describe the problem you're observing:

This is a continuation of bug report #13322. I'm closing the other bug report as the original submission issue is fixed, but the fix involved changing some driver properties in the JDBC driver to show the auto generated / auto increment property information. The problem however, is that now messed up how DBeaver is interpreting column metadata and I can no longer see the column name + column label/description below the column name in the data editor grid.

#### Steps to reproduce, if exist:

In bug report #13322 I found how to get additional metadata property information to show the auto generated / auto increment property info to get passed back.

The problem however is that this seems to have broken the column label's from being displayed.

Right now I have the following set in the JDBC driver:

metadata source: 1
translate binary: true
date format: iso
time format: iso
extended metadata: true

And the following items marked in the UI:

![image](https://user-images.githubusercontent.com/6610039/127174858-2d7818af-b71d-4304-87c1-c1e44d5aca7c.png)

![image](https://user-images.githubusercontent.com/6610039/127174888-7a78e375-da6c-4dd2-b850-97ce4b15b8e3.png)

However when I query a table I get this:

![image](https://user-images.githubusercontent.com/6610039/127175081-a1704d0e-3ccc-426d-9ce3-304d2ce43427.png)

I do see the metadata panel has the column descriptions however, they just are not displaying underneath the column in the grid:

![image](https://user-images.githubusercontent.com/6610039/127175193-71cfb914-da3a-4961-9459-b8e8ede89bfb.png)

There were some customizations around column headers performed in a very old DBeaver PR for IBMi / AS400 connections way back on this #1105. Those customizations were done around the old style metadata source: 0 flag, and I wonder if perhaps those are interfering here.

My main question is, how to get the column labels that DBeaver is clearly able to retrieve with the above JDBC settings show underneath the columns in the grid. They should be showing as the checkbox in the second screenshot is marked.

Ideally this is what I am after in the data grid, which I can obtain with the following settings:

metadata source: 0
translate binary: true
date format: iso
time format: iso
extended metadata: false

![image](https://user-images.githubusercontent.com/6610039/127176616-aa86defa-4fd7-4cc5-a5f2-766a1431f963.png)

The problem with the above settings however is that metadata source: 0 appears to be the old style of getting metadata and is no longer the default (the default is now 1 on later versions of the IBMi).  Additionally metadata source: 0 does not allow you to view any of the extra information about a column such as it's auto-increment / identity status.

**Comment by bdietz400** [comment]
To assist in testing on an IBM i server, you may be able to get a free IBM i account at https://pub400.com/.

**Comment by Matvey16** [comment]
#541 Can have useful information about column metadata

**Comment by LonwoLonwo** [comment]
Hello @xantari 

Now we have a test environment, and I can't reproduce your issue, unfortunately.

I set:
metadata source: 1
translate binary: true
extended metadata: true

But I can see columns and tables remarks.

Also, did you try to change the "remarks" driver setting? 
https://www.ibm.com/docs/en/i/7.4?topic=ssw_ibm_i_74/rzahh/javadoc/com/ibm/as400/access/doc-files/JDBCProperties.html
Maybe it can help?

![2022-04-01 10_56_47-Connection _PUB400 COM 1 1_ configuration](https://user-images.githubusercontent.com/45152336/161221668-7b942697-1a64-40e5-9b45-bac2aec714e5.png)

As I understand, "metadata source" read labels, not comments https://sourceforge.net/p/jt400/bugs/384/#a86d

**Comment by xantari** [comment]
Hi,

Let me give you example.

1. Create the following source file on your system. [VUECANDIDATEDEMOGRAPHICSHISTORY.txt](https://github.com/dbeaver/dbeaver/files/8411747/VUECANDIDATEDEMOGRAPHICSHISTORY.txt)
2. Set the driver properties as follows:
metadata source: 1
translate binary: true
extended metadata: true
remarks: system
3. Set DBeaver to show column comments:
![image](https://user-images.githubusercontent.com/6610039/161595885-01f7541a-f843-4b07-bf4e-6aafd4dea882.png)
![image](https://user-images.githubusercontent.com/6610039/161595923-3c93f8d8-6985-471e-adfc-153146eebcfa.png)
4. Now do "select * from ARRTFLIB.VUECANDIDATEDEMOGRAPHICSHISTORY". I get the following **with missing comment information underneath the column**:
![image](https://user-images.githubusercontent.com/6610039/161596089-95e98fc3-c9c4-4500-a719-c4a8f1034590.png)

Only way to fix the missing column information is to go back to driver properties and set "metadata source: 0".

Once you set "metadata source: 0" you then get properly working DBeaver when you do a "select * from ARRTFLIB.VUECANDIDATEDEMOGRAPHICSHISTORY" as shown here:
![image](https://user-images.githubusercontent.com/6610039/161596531-ffc30ae2-f4e1-4ab1-aa35-4f085539b51f.png)

But now, DBeeaver is missing column information details from table with "metadata source: 0":

![image](https://user-images.githubusercontent.com/6610039/161596879-898b1817-2746-4f0b-af71-bf72bcf66b8d.png)

Somehow column text is broken in DBeaver when "metadata source: 1" is set.

**Comment by LonwoLonwo** [comment]
Ok, I see that you are using labels for your columns.

As some workaround, for now, I suggest you use COMMENTS for columns, not just labels. 

![2022-04-05 10_01_03-DBeaver Ultimate 22 0 2 - _PUB400 COM 1_Script-145](https://user-images.githubusercontent.com/45152336/161703367-32665a2e-f222-475d-bf2e-61d70027e20d.png)

Also, please, can you tell me - how do you use DBeaver with "metadata source: 0"? In this case, the driver does not return schemas, as I can see. Is it ok for you?

**Comment by xantari** [comment]
Yes metadata source: 0 works for me, but is missing a few features such as showing identity column details.

**Comment by Starcomfr** [comment]
Yes but unfortunately this request has the label "Planned" but no more due date because @tati-kru removed the milestonesfor 22.2.3 instead of modified it.
